enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult/InteractiveQuestionResult.tsx

```diff
@@ -84,39 +68,58 @@ export const InteractiveQuestionResult = ({
       style={style}
     >
-      <Group className={InteractiveQuestionS.TopBar} position="apart" p="md">
-        <InteractiveQuestion.BackButton />
-        {withTitle && (customTitle ?? <InteractiveQuestion.Title />)}
-        <Group spacing="xs">
-          {withResetButton && <InteractiveQuestion.ResetButton />}
-          <InteractiveQuestion.ChartTypeDropdown />
-          <InteractiveQuestion.FilterDropdown />
-          <InteractiveQuestion.SummarizeDropdown />
-          <InteractiveQuestion.BreakoutDropdown />
-          <InteractiveQuestion.EditorButton
-            isOpen={questionView === "editor"}
-            onClick={() =>
-              setQuestionView(
-                questionView === "editor" ? "visualization" : "editor",
-              )
-            }
-          />
-
+      <Stack className={InteractiveQuestionS.TopBar} spacing="sm" p="md">
+        <Group position="apart" align="flex-end">
+          <Group spacing="xs">
+            <InteractiveQuestion.BackButton />
+            <ResultTitle
+              withResetButton={withResetButton}
+              customTitle={customTitle}
+              withTitle={withTitle}
+            />
+          </Group>
           {isSaveEnabled && !isSaveModalOpen && (
             <InteractiveQuestion.SaveButton onClick={openSaveModal} />
           )}
         </Group>
-      </Group>
 
-      <Box className={InteractiveQuestionS.Main} p="md" w="100%" h="100%">
-        <Box className={InteractiveQuestionS.Content}>
-          <ContentView
-            questionView={questionView}
-            onReturnToVisualization={() => setQuestionView("visualization")}
+        <Group
+          position="apart"
+          p="sm"
+          bg="var(--mb-color-background-disabled)"
+          style={{ borderRadius: "0.5rem" }}
+        >
+          <Group spacing="xs">
+            <InteractiveQuestion.ChartTypeDropdown />
+            <Divider
+              mx="xs"
+              orientation="vertical"
+              // we have to do this for now because Mantine's divider overrides this color no matter what
+              color="var(--mb-color-border) !important"
+            />
+            <InteractiveQuestion.FilterDropdown />
+            <InteractiveQuestion.SummarizeDropdown />
+            <InteractiveQuestion.BreakoutDropdown />
+          </Group>
+          <InteractiveQuestion.EditorButton
+            isOpen={isEditorOpen}
+            onClick={toggleEditor}
           />
+        </Group>
+      </Stack>
+
+      <Box className={InteractiveQuestionS.Main} p="sm" w="100%" h="100%">
+        <Box className={InteractiveQuestionS.Content}>
+          {match<boolean>(isEditorOpen)
+            .with(true, () => (
+              <InteractiveQuestion.Editor onApply={closeEditor} />
+            ))
+            .with(false, () => (
+              <InteractiveQuestion.QuestionVisualization height="100%" />
+            ))
+            .exhaustive()}
```

Comment:
For boolean matching a ternary would be enough

```suggestion
          {isEditorOpen ? (
             <InteractiveQuestion.Editor onApply={closeEditor} />
           ) : (
             <InteractiveQuestion.QuestionVisualization height="100%" />
           )}
```

Reply:
I got too excited about `ts-pattern` 😂